### PR TITLE
feat: add support for Slack slash commands

### PR DESCRIPTION
**Pull request by tw0517tw**
~need example~

Changes:
1. slack slash command support including
  * router
  * example
  * documentation

### CODE REVIEW

File: src/bot/SlackConnector.js [hunk 1]
```diff
@@ -62,7 +63,7 @@ export default class SlackConnector implements Connector<SlackRequestBody> {
     } else if (body.payload && typeof body.payload === 'string') {
       return (JSON.parse(body.payload): InteractiveMessageEvent);
     }
-    // for RTM WebSocket messages
+    // for RTM WebSocket messages and Slach Command messages
```
Comment (tigercosmos) [inline src/bot/SlackConnector.js]: Slach => Slash?

**Comment by lkostrowski** [comment]
how about that?

File: examples/slack-slash-command/README.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+## Idea of this Example
+
+This example is a simple bot running on [Slack](https://slack.com/).
+For more information, check our [Slack guides](https://bottender.js.org/docs/channel-slack-setup).
+
+## Related Examples
+
+- [messenger-hello-world](../messenger-hello-world)
+- [slack-hello-world](../slack-hello-world)
```
Comment (chentsulin) [inline examples/slack-slash-command/README.md]: We should only keep `slack-hello-world` example here.

File: packages/bottender/src/slack/SlackEvent.ts [hunk 3]
```diff
@@ -286,6 +300,22 @@ export default class SlackEvent implements Event<SlackRawEvent> {
...
+   */
+  get isSlashCommand(): boolean {
+    return !!(this._rawEvent as SlashCommandEvent).command;
+  }
+
+  /**
+   * The slash command name.
+   *
+   */
+  get slashCommand(): string | null {
```
Comment (chentsulin) [inline packages/bottender/src/slack/SlackEvent.ts]: We should use `command` and `isCommand` consistently. It seems that slack doesn't use slash in their code or field names.

File: packages/bottender/src/slack/SlackEvent.ts [hunk 4]
```diff
@@ -200,7 +216,7 @@ export default class SlackEvent implements Event<SlackRawEvent> {
    *
    */
   get isText(): boolean {
-    return this.isMessage;
+    return this.isMessage || !!(this._rawEvent as SlashCommandEvent).text;
```
Comment (tw0517tw) [inline packages/bottender/src/slack/SlackEvent.ts]: I wonder if we want to treat slash command as text event.
Considering `/command` and `/command some args`, why the latter should be treated as text event and the former should not?

Comment (darkbtf) [reply packages/bottender/src/slack/SlackEvent.ts]: I agree.
It's kind of confusing the arguments string is in `event.text`

File: packages/bottender/src/slack/__tests__/SlackEvent.spec.ts [hunk 5]
```diff
@@ -551,6 +551,21 @@ const interactiveMessageEvent = {
   triggerId: '274927463524.5223114719.95a5b9f6d3b30dc7e07dec6bfa4610e5',
 };
 
+const slashCommandMessageEvent = {
```
Comment (tw0517tw) [inline packages/bottender/src/slack/__tests__/SlackEvent.spec.ts]: how about adding a fixture for slash command without arguments?

Comment (chentsulin) [reply packages/bottender/src/slack/__tests__/SlackEvent.spec.ts]: @darkbtf

Comment (darkbtf) [reply packages/bottender/src/slack/__tests__/SlackEvent.spec.ts]: ok, I'll add that test case

**Pull request merged**